engine/src/test/scala/cromwell/engine/workflow/lifecycle/execution/EngineJobExecutionActorSpec.scala

```diff
@@ -0,0 +1,192 @@
+  val mockBackendProps = Props(new Actor {
+    def receive = {
+      case x => backendProbe.ref forward x
+    }
+  })
+  val ejeaParent = TestProbe()
+  val factory = new BackendLifecycleActorFactory {
+    override def workflowInitializationActorProps(workflowDescriptor: BackendWorkflowDescriptor, calls: Seq[Call]): Option[Props] = None
+    override def jobExecutionActorProps(jobDescriptor: BackendJobDescriptor, initializationData: Option[BackendInitializationData]): Props = mockBackendProps
+    override def expressionLanguageFunctions(workflowDescriptor: BackendWorkflowDescriptor, jobKey: BackendJobDescriptorKey, initializationData: Option[BackendInitializationData]): WdlStandardLibraryFunctions = {
+      NoFunctions
+    }
+  }
+
+  def buildEJEA() = new TestFSMRef[EngineJobExecutionActorState, EngineJobExecutionActorData, EngineJobExecutionActor](system, EngineJobExecutionActor.props(
+    WorkflowExecutionActorData(descriptor, ExecutionStore(Map.empty), Map.empty, OutputStore(Map.empty), restarting = false),
+    factory,
+    None
+  ), ejeaParent.ref, s"EngineJobExecutionActorSpec-$workflowId")
+
+  "EngineJobExecutionActorSpec" should {
+    "send a Job SucceededResponse if the job is already complete and successful" in {
+      val ejea = buildEJEA()
+      ejea.setState(CheckingJobStatus)
+      val returnCode: Option[Int] = Option(0)
+      val jobOutputs: JobOutputs = Map.empty
+
+      ejea ! JobComplete(mock[BackendJobDescriptorKey], JobResultSuccess(returnCode, jobOutputs))
+
+      ejeaParent.expectMsgPF(awaitTimeout) {
+        case response: SucceededResponse =>
+          response.returnCode shouldBe returnCode
+          response.jobOutputs shouldBe jobOutputs
+      }
+
+      ejea.stop()
+    }
+
+    "send a Job FailedNonRetryableResponse if the job is already complete and failed" in {
+      val ejea = buildEJEA()
+      ejea.setState(CheckingJobStatus)
+      val returnCode: Option[Int] = Option(1)
+      val reason: Throwable = new Exception("something horrible happened...")
+
+      ejea ! JobComplete(mock[BackendJobDescriptorKey], JobResultFailure(returnCode, reason))
+
+      ejeaParent.expectMsgPF(awaitTimeout) {
+        case response: FailedNonRetryableResponse =>
+          response.returnCode shouldBe returnCode
+          response.throwable shouldBe reason
+      }
+
+      ejea.stop()
+    }
+
+    "send a RecoverJobCommand to the backend if the job is not complete" in {
+      val ejea = buildEJEA()
+      ejea.setState(CheckingJobStatus, EngineJobExecutionActorData(true))
+      val task = mock[Task]
+      task.declarations returns Seq.empty
+
+      val jobKey = BackendJobDescriptorKey(Call(None, "wf.call", task, Set.empty, Map.empty, None), None, 1)
+      ejea ! JobNotComplete(jobKey)
+
+      backendProbe.expectMsg(awaitTimeout, RecoverJobCommand)
+
+      ejea.stop()
+    }
+
+    // This test is bit redundant with the one above, hoever this one tests the behaviour when getting a Restart message
```

Comment:
Bizarrely, there are only 11 flags total on the cheat sheet. And two of them we : gb: :gb: and : uk: :uk: